.github/workflows/consumer_contract_tests.yml

```diff
@@ -121,50 +122,72 @@ jobs:
           distribution: 'temurin'
       - name: Run consumer tests
         run: ./gradlew pactTests
-      - name: Output consumer contract as non-breaking base64 string
-        id: encode-pact
+      - name: Locate pact files
+        id: locate-pacts
         run: |
-          NON_BREAKING_B64=$(cat service/build/pacts/bpm-sam.json | base64 -w 0)
-          echo "pact-b64=${NON_BREAKING_B64}" >> $GITHUB_OUTPUT
+          pactPaths=$(find "$PACTS_OUTPUT_DIR" -type f -name "*.json" | jq -cnR "[inputs]")
+          echo "pact-paths=$pactPaths" >> $GITHUB_OUTPUT
+      - name: Upload pact files to artifact
+        id: upload-pacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ env.PACTS_ARTIFACT }}
+          path: ${{ env.PACTS_OUTPUT_DIR }}
+          retention-days: 1
 
   publish-contracts:
     runs-on: ubuntu-latest
     needs: [ bump-check, init-github-context, bpm-consumer-contract-tests, regulated-tag-job ]
+    strategy:
+      matrix:
```

Comment:
Why does this need matrix? Matrix has caused problems in the past with incorrect github actions runs getting linked, and I don't see that we are iterating over anything.

Reply:
Unless `pact-path` is more than a single path?

Reply:
Yes, looking at the actual ticket for this change I see it is about publishing multiple pacts. So I understand now the matrix call.